### ISSUE TITLE
Segfaults with primitive types and virtual method calls

### ISSUE THREAD

**Issue by kyonifer** [JetBrains/kotlin-native#558]
I was running into some issues implementing a type hierarchy and boiled down a test case that reproduces the issue for me. The following code complies cleanly but segfaults at runtime for me, on kotlin-native 0.1 and 0.2:

```kotlin

interface Matrix<T> {
    operator fun set(i: Int, v: T)
    operator fun set(i: Int, j: Int, v: T)
    operator fun get(i: Int, j: Int): T
    operator fun get(i: Int): T
}
// Things work if this intermediate class isnt here, and DoubleMatrix descends from Matrix directly
abstract class MatrixBase<T>: Matrix<T> {}

abstract class DoubleMatrix : MatrixBase<Double>() {
}

class DefaultDoubleMatrix (val rows: Int,
                           val cols: Int): DoubleMatrix() {
    val storage = DoubleArray(rows*cols)

    override fun set(i: Int, v: Double) {
        storage[i] = v
    }
    override fun set(i: Int, j: Int, v: Double) {
        storage[cols*i+j] = v
    }
    override fun get(i: Int, j: Int): Double {
        return storage[cols*i+j]
    }
    override fun get(i: Int): Double
            = storage[i]
}


// Things work if return type is DefaultDoubleMatrix instead
fun makeDoubleMatrix(): DoubleMatrix = DefaultDoubleMatrix(3,3)

fun main(args: Array<String>) {
    val a = makeDoubleMatrix()
    // This causes the println below to be 0, then segfaults
    a[0] = 2.4
    // This segfaults
    a[0,0]=2.5

    println("${a[0]}")
}
```

**Comment by vvlevchenko** [comment]
thank you for reporting.

**Comment by homuroll** [comment]
Fixed in `master` and `v0.2-fixes`.
